### PR TITLE
Answer modal

### PR DESCRIPTION
**Pull request by lawsun03**
Implemented the image upload functionality into the AnswerModal while refactoring the QuestionModal validation system. In order to render the Questions and Answers list immediately after a post request is made for either an answer or question to be added, I shifted state components into higher components to pass down get request functions as props to the Modal components. 

All modal components seem to be functioning but planing to refactor the image upload process of the answers modal to be more user friendly. Need to also implement the collapsing of questions  list when there is a product with no questions.

### CODE REVIEW

File: client/src/Components/QuestionsContainer/QuestionsList/AnswersList/AnswersList.jsx [hunk 1]
```diff
@@ -85,7 +55,7 @@ const AnswersList = (props) => {
             <Col>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</Col>
```
Comment (ggbbi) [inline client/src/Components/QuestionsContainer/QuestionsList/AnswersList/AnswersList.jsx:55]: Might want to use `style={{ float: right }}` or something similar? not sure if this is best practice

File: client/src/Components/QuestionsContainer/QuestionsList/AnswersList/AnswersList.jsx [hunk 2]
```diff
@@ -26,41 +26,11 @@ const AnswersList = (props) => {
...
-  }
-
-  if (answersListData.length > 2 && renderedAnswersListData.length >= 2) {
+  if (
+    props.answersListData.length > 2 &&
+    props.renderedAnswersListData.length >= 2
+  ) {
     // conditionally renders whether there will be a 'SEE MORE ANSWERS' button or not
     return (
       <>
```
Comment (ggbbi) [inline client/src/Components/QuestionsContainer/QuestionsList/AnswersList/AnswersList.jsx:36]: You replaced these brackets with `<React.Fragment>` in another file, maybe change for consistency?

File: client/src/Components/QuestionsContainer/QuestionsList/QuestionsListEntry/AnswerModal/AnswerModal.jsx [hunk 3]
```diff
@@ -31,91 +32,105 @@ const AnswerModal = (props) => {
...
+    const uploaders = files.map((file) => {
+      const formData = new FormData()
+      formData.append("file", file)
+      // formData.append("tags", `codeinfuse, medium, gist`)
+      formData.append("upload_preset", "wcjmqzvr")
+      // formData.append("api_key", "353977359896185")
+      // formData.append("timestamp", (Date.now() / 1000) | 0)
+
+      axios
+        .post("https://api.cloudinary.com/v1_1/dqab317k6/image/upload", formData)
```
Comment (ggbbi) [inline client/src/Components/QuestionsContainer/QuestionsList/QuestionsListEntry/AnswerModal/AnswerModal.jsx:122]: Looks like this is used to create a URL that can be stored/used later? I might use this too if it works!

File: client/src/Components/QuestionsContainer/QuestionsList/QuestionsListEntry/QuestionsListEntry.jsx [hunk 4]
```diff
@@ -46,6 +60,28 @@ const QuestionsListEntry = (props) => {
...
+  }
+
+  const getAnswersRenderData = () => {
+    setRenderedAnswersListData(answersListData.slice(0, answersCount))
+  }
+
+  const ShowMoreOrCollapse = () => {
+    if (answersCount === 2) {
+      return (
+        <SmallBold onClick={() => setAnswersCount(12)}>SEE MORE ANSWERS</SmallBold>
```
Comment (ggbbi) [inline client/src/Components/QuestionsContainer/QuestionsList/QuestionsListEntry/QuestionsListEntry.jsx:76]: Same about the `<a>` tag here, for automatic link styling

File: client/src/Components/QuestionsContainer/QuestionsContainer.jsx [hunk 5]
```diff
@@ -1,34 +1,38 @@
-import React, { useState } from "react"
+import React, { useState, useEffect } from "react"
```
Comment (ggbbi) [inline client/src/Components/QuestionsContainer/QuestionsContainer.jsx:1]: It doesn't look like `useEffect` is being used here, can be deleted?

**Review by ggbbi — APPROVED** [review]
Nice! I made comments relating to styling & trying to align the styles across app modules -- would be great if you could address those.

File: client/src/Components/QuestionsContainer/QuestionsList/QuestionModal/QuestionModal.jsx [hunk 6]
```diff
@@ -128,27 +99,26 @@ const QuestionModal = () => {
...
-          <Form>
+
             <Form.Group className="mb-3" controlId="QuestionForm.Email">
               <Form.Label>Email address:</Form.Label>
               <Form.Control
+                required
                 onChange={handleEmail}
-                type="email"
-                placeholder="Why did you like the product or not?"
+                pattern="[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{1,63}$"
```
[the remark this answers is not in the document]
Comment (lawsun03) [reply client/src/Components/QuestionsContainer/QuestionsList/QuestionModal/QuestionModal.jsx:108]: I noticed that type="email", would verify the section for values such as "characters@something" but using this pattern, it requires the section to also include a dot and something after the dot for example: "characters@something.com"

File: client/src/Components/QuestionsContainer/QuestionsList/QuestionModal/QuestionModal.jsx [hunk 7]
```diff
@@ -128,27 +99,26 @@ const QuestionModal = () => {
...
+                placeholder="Example: jackson22@email.com"
                 value={email}
               />
               <UnderText>
```
[the remark this answers is not in the document]
Comment (lawsun03) [reply client/src/Components/QuestionsContainer/QuestionsList/QuestionModal/QuestionModal.jsx:112]: Got it! will make those changes.

File: client/src/Components/QuestionsContainer/QuestionsList/QuestionsListEntry/AnswerModal/AnswerModal.jsx [hunk 8]
```diff
@@ -31,91 +32,105 @@ const AnswerModal = (props) => {
...
+          urls.push(fileURL)
+          console.log(data)
+        })
+    })
+    setPhotos(urls)
+  }
+
   return (
     <>
       <SmallUnderline onClick={handleShow}>Add Answer</SmallUnderline>
```
[the remark this answers is not in the document]
Comment (lawsun03) [reply client/src/Components/QuestionsContainer/QuestionsList/QuestionsListEntry/AnswerModal/AnswerModal.jsx:136]: Got it! I'll address this change soon!

File: client/src/Components/QuestionsContainer/QuestionsList/QuestionsListEntry/AnswerModal/AnswerModal.jsx (hunk 3 above)
Comment (lawsun03) [reply client/src/Components/QuestionsContainer/QuestionsList/QuestionsListEntry/AnswerModal/AnswerModal.jsx:122]: It works great! You can use the same format to upload images if you'd like. I created an account on cloudinary for it so if you want you could just use that same endpoint.

**Pull request merged**